jazz_environment-event-handler/test/test.js

```diff
@@ -181,11 +277,169 @@ describe('jazz environment handler tests: ', () => {
 		});
 	});
+
+	it('Verify processEachEvent for CREATE_BRANCH event failed', () => {
+		let event = require('./CREATE_BRANCH');
+		const statusCode = testPayloads.createBranchSuccess.statusCode;
+		let requestPromiseStub = sinon.stub(request, "Request", (obj) => {
+			return obj.callback(null, testPayloads.createBranchError, testPayloads.createBranchError.body);
+		});
+
+		index.processItem(event.Item, configData, authToken)
+		.catch((res) => {
+			sinon.assert.calledOnce(requestPromiseStub);
+			requestPromiseStub.restore();	
+			expect(res.details).to.include('error');
+		});
+	});
+
+	it('Verify getEnvironmentLogicalId returns a valid logical Id for a branch', () => {
+		let environmentPayload = testPayloads.environmentPayload
+		let logical_id = "6knr9d33tt-dev";
+
+		let requestPromiseStub = sinon.stub(request, "Request", (obj) => {
+			return obj.callback(null, testPayloads.getEnvironmentLogicalId, JSON.stringify(testPayloads.getEnvironmentLogicalId.body));
+		});
+
+		index.getEnvironmentLogicalId(environmentPayload, configData, authToken)
+		.then((res) => {
+			sinon.assert.calledOnce(requestPromiseStub);
+			requestPromiseStub.restore();
+			expect(res).to.eql(logical_id);
+		});
+	});
+
+	it('Verify getEnvironmentLogicalId throws error when status code is 400', () => {
+		let environmentPayload = testPayloads.environmentPayload
+		const statusCode = testPayloads.getEnvironmentLogicalId.statusCode;
+		testPayloads.getEnvironmentLogicalId.statusCode = 400;
+		let requestPromiseStub = sinon.stub(request, "Request", (obj) => {
+			return obj.callback(null, testPayloads.getEnvironmentLogicalId, JSON.stringify(testPayloads.getEnvironmentLogicalId.body));
+		});
+
+		index.getEnvironmentLogicalId(environmentPayload, configData, authToken)
+		.catch((res) => {
+			sinon.assert.calledOnce(requestPromiseStub);
+			requestPromiseStub.restore();
+			testPayloads.getEnvironmentLogicalId.statusCode = statusCode
+			expect(res.error).to.eql('Could not get environment Id for service and domain');
+		});
+	});
+
+	it('Verify processEventUpdateEnvironment rejects with statusCode of 400', () => {
+		let environmentPayload = testPayloads.environmentPayload;
+		environmentPayload.friendlyName = "FriendlyName";
+		environmentPayload.logicalId = "6knr9d33tt-dev";
+
+		let requestPromiseStub = sinon.stub(request, "Request", (obj) => {
+			return obj.callback(null, testPayloads.processEventUpdateEnvironmentError, JSON.stringify(testPayloads.processEventUpdateEnvironmentError.body));
+		});
+
+		index.processEventUpdateEnvironment(environmentPayload, configData, authToken)
+		.catch(res => {
+			sinon.assert.calledOnce(requestPromiseStub);
+			requestPromiseStub.restore();
+			expect(res.details).to.eql('Error');
+		})
+	});
+
+	it('Verify processEventCreateBranch rejects with statusCode of 400', () => {
+		let environmentPayload = testPayloads.environmentPayload;
+		environmentPayload.logicalId = "6knr9d33tt-dev";
```

Comment:
It is not clear what makes _this_ unit test case reject with status code of 400? Can you comment or make it more explicit as to why this is returning 400?